### PR TITLE
modified for PageObjectPattern model

### PR DESCRIPTION
**Pull request by aga2zak**

### CODE REVIEW

File: ITCrowdProject/src/main/java/pl/infoshare/dataModels/Bag.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
+package pl.infoshare.dataModels;
+
+public class Bag  {
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/dataModels/Bag.java]: watch out for possible conflicts - I think this code has been a part of another pull request

File: ITCrowdProject/src/main/java/pl/infoshare/elements/Button.java [hunk 2]
```diff
@@ -33,6 +33,13 @@ public Button safeClick(){
     public Button clickWithJs(){
         JavascriptExecutor runner = (JavascriptExecutor) driver;
         String script = "arguments[0].click()";
+       // String script = "arguments[0].scrollIntoView()";
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/elements/Button.java]: looks like it's unnecessary - remove that line

File: ITCrowdProject/src/main/java/pl/infoshare/generators/BagGenerator.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    private static Bag chicVintageDeVille = new Bag( "Handbags", "Chic","Chic vintage DeVille", true, 78, 1, 17, 28, 4);
+    private static Bag vintageChicHandbag = new Bag( "Handbags", "Chic","Vintage chic handbag", true, 68, 2, 17, 28, 5);
+    private static Bag vintageCourierBag = new Bag( "Handbags", "Courier","Vintage courier bag", true, 60, 2, 18, 22, 10);
+    private static Bag retroStyleBag = new Bag( "Beach bags", "Retro","Retro style bag", true, 48, 1, 20, 30, 4);
+    private static Bag vintageBeachBag = new Bag( "Beach bags", "Beach","Vintage beach bag", true, 60, 1, 20, 30, 8);
+    private static Bag vintageLaptopBag = new Bag( "Laptop bags", "Vintage","Vintage laptop bag", true, 78, 2, 18, 20, 4);
+    private static Bag vintageBagWithLeatherBands = new Bag( "Bags", "Retro","Vintage bag with leather bands", true, 68, 1, 19, 27, 4);
+    private static List<Bag> bagList;
+
+    static   {
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/generators/BagGenerator.java]: what is that bit here?

File: ITCrowdProject/src/main/java/pl/infoshare/pages/CatalouqePage.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    public void checkout(){
+        this.shoppingCart = new Button(this.driver, this.shippingCartLink);
+        this.shoppingCart.safeClick();
+        this.chackoutLink= By.cssSelector("#miniCartDetails > li.checkout-bg > a");
+        this.checkout= new Button(this.driver, this.chackoutLink);
+        checkout.clickWithJs();
+    }
+
+    public void getRandomBag(String name){
+        bag = By.cssSelector("#productsContainer [item-name='"+name+"'] .addToCart");
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/pages/CatalouqePage.java]: this is really cool!!! respect

File: ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    private By firstNameInput=By.cssSelector("input[name='customer.billing.firstName']");
+    private TextInput firstName;
+    private By lastNameInput=By.cssSelector("input[name='customer.billing.lastName']");
+    private TextInput lastName;
+    private By streetInput=By.cssSelector("input[name='customer.billing.address']");
+    private TextInput street;
+    private By cityInput=By.cssSelector("input[name='customer.billing.city']");
+    private TextInput city;
+    private By countrySelect;
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java]: these bys do not change, so I guess you can put them at the top of the class, just like private By cityInput

File: ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public void sectionFullName(RegisteredUser user){
+        this.firstName.sendKeys(user.getFirstname());
+        this.lastName.sendKeys(user.getLastname());
+        this.street.sendKeys(user.getAddress().getStreetAddress());
+        this.city.sendKeys(user.getAddress().getCity());
+    }
+
+    public void sectionCountryState(RegisteredUser user){
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java]: I'd name these methods starting with verbs, like "fillInSectionCountryState", but that's your call ;)

File: ITCrowdProject/src/main/java/pl/infoshare/pages/MainPage.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    private By handbagsLink;
+    private Button handbagsButton;
+    private By categoryLink;
+    private Button categoryButton;
+
+    public MainPage(WebDriver driver){
+        super(driver);
+        this.url = "http://demo.shopizer.com:8080/shop/";
+        this.driver.get(this.url);
+       // this.handbagsButton = new Button(this.driver, this.handbagsLink);
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/pages/MainPage.java]: remove if it's not needed

File: ITCrowdProject/src/main/java/pl/infoshare/pages/components/CategoriesMenu.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import pl.infoshare.elements.Button;
+import pl.infoshare.pages.BasePage;
+
+public class CategoriesMenu {
+    private String url;
+
+    public String switchToPage(String categories) {
+        switch (categories) {
```
Comment (katarzynaboguslawska) [inline ITCrowdProject/src/main/java/pl/infoshare/pages/components/CategoriesMenu.java]: really nice! If you want to be even more pro, checkhow to put these categories into a reusable enum :)

File: ITCrowdProject/src/main/java/pl/infoshare/elements/Button.java (hunk 2 above)
Comment (aga2zak) [reply ITCrowdProject/src/main/java/pl/infoshare/elements/Button.java]: Yes, It's deleted.

File: ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java (hunk 5 above)
Comment (aga2zak) [reply ITCrowdProject/src/main/java/pl/infoshare/pages/CheckoutPage.java]: I've changed it.

File: ITCrowdProject/src/main/java/pl/infoshare/pages/MainPage.java (hunk 7 above)
Comment (aga2zak) [reply ITCrowdProject/src/main/java/pl/infoshare/pages/MainPage.java]: I've removed it.

**Pull request closed without merging**